nb2/service/dtos.py

```diff
@@ -0,0 +1,48 @@
+from dataclasses import dataclass
+
+
+class BaseDTO:
+    REQUIRED_FIELDS = []
+
+    def get_empty_fields(self):
+        """
+        Return a list of field names that have a value of None.
+        """
+        return [f for f in self.__dict__ if not self.__dict__[f]]
+
+    def get_empty_required_fields(self):
+        """
+        Return a list of field names that have a value of None
+        and are listed in REQUIRED_FIELDS.
+        """
+        empty_fields = self.get_empty_fields()
+        return [f for f in empty_fields if f in self.REQUIRED_FIELDS]
+
+
+@dataclass
```

Comment:
I really like what you did here! 🎉

Reply:
Thanks! I was going to go with namedtuples when I stumbled upon this cutting edge Python technology!